onnx/onnx.proto

```diff
@@ -75,15 +76,25 @@ message AttributeProto {
   repeated GraphProto graphs = 11;    // list of graph
 }
 
+// Defines information on value, including the name, the type, and
+// the shape of the value.
+message ValueInfoProto {
+  // This field MUST be present in this version of the IR.  
+  optional string name = 1;     // namespace Value
+  // This field MUST be present in this version of the IR.  
+  optional TypeProto type = 2;
+  optional TensorShapeProto shape = 3;
```

Comment:
I have been thinking about this some more, and I don't think shape should be besides the TypeProto. The shape specification can depend on whether or not you have a dense or sparse tensor; for example, PyTorch sparse tensors have both sparse dimensions and dense dimensions, and that should be required in the shape.